core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -286,7 +279,7 @@ private ClusterState applyRequest(ClusterState currentState, PutMappingClusterSt
                             for (ObjectCursor<MappingMetaData> mapping : indexMetaData.getMappings().values()) {
                                 String parentType = newMapper.parentFieldMapper().type();
                                 if (parentType.equals(mapping.value.type()) &&
-                                        indexService.mapperService().getParentTypes().contains(parentType) == false) {
+                                    mapperService.getParentTypes().contains(parentType) == false) {
```

Comment:
can you indent one level more? otherwise it's not obvious what belongs to the if statement and what belongs to the inner block